crates/contract/src/call.rs

```diff
@@ -271,8 +271,10 @@ impl<P: TempProvider, D: CallDecoder> CallBuilder<P, D> {
     }
 
     /// Uses a Legacy transaction instead of an EIP-1559 one to execute the call
-    pub fn legacy(self) -> Self {
-        todo!()
+    pub fn legacy(mut self) -> Self {
+        self.request = self.request.max_fee_per_gas(U256::ZERO);
+        self.request = self.request.max_priority_fee_per_gas(U256::ZERO);
```

Comment:
these should be set to None instead
this also needs to remove any blob fields

Reply:
> these should be set to None instead this also needs to remove any blob fields

thanks!